### ISSUE TITLE
Add ResetWorkflowExecution support in TestWorkflowEnvironment

### ISSUE THREAD

**Issue by tsurdilo** [temporalio/sdk-java#1093]
Currently when you try to use ResetWorkflowExecution with client from TestWorkflowEnvironment you get:

       io.grpc.StatusRuntimeException: UNIMPLEMENTED: Method temporal.api.workflowservice.v1.WorkflowService/ResetWorkflowExecution is unimplemented

Would be nice to add support for it in the future. Thanks.

**Comment by Spikhalskiy** [comment]
You can use dockerized temporal for tests that have to cover reset of the workflow. 
We may consider implementing some simplified version of a reset in the test server in the future.

**Comment by tsurdilo** [comment]
Yeah, forgot to mention this is when setUseExternalService is not set to true

**Comment by parth-kulkarni-db** [comment]
Is there any update if this is added in Test Server in Apr 23?
